File: solutions/3111. Minimum Rectangles to Cover Points/3111.cpp
// Time complexity: O(sort)
// Space complexity: O(1)
class Solution {
 public:
  int minRectanglesToCoverPoints(vector<vector<int>>& points, int w) {
    ranges::sort(points);
    int ans = 0;
    int idx = -1;

    for (const vector<int>& point : points)
      if (point[0] > idx) {
        ++ans;
        idx = point[0] + w;
      }

    return ans;    
  }
};

File: solutions/3111. Minimum Rectangles to Cover Points/3111.py
# Time complexity: O(sort)
# Space complexity: O(1)
class Solution:
  def minRectanglesToCoverPoints(self, points: List[List[int]], w: int) -> int:
    points.sort()

    ans = 0
    idx = -1

    for point in points:
      if point[0] > idx:
        ans += 1
        idx = point[0] + w

    return ans    